features/fixtures/mazerunner/src/main/java/com/bugsnag/android/mazerunner/MainActivity.kt

```diff
@@ -30,33 +62,21 @@ class MainActivity : Activity() {
         }, 1)
     }
 
-    private fun loadScenario(configuration: Configuration): Scenario {
-        val eventType = intent.getStringExtra("EVENT_TYPE")
-        val eventMetaData = intent.getStringExtra("EVENT_METADATA")
+    private fun loadScenario(configuration: Configuration, eventType: String, eventMetaData: String): Scenario {
         Log.d("Bugsnag", "Received test case, executing " + eventType)
+        Log.d("Bugsnag", "Received metadata: " + eventMetaData)
 
+        this.intent.putExtra("eventMetaData", eventMetaData)
```

Comment:
This is necessary in scenarios that require metadata when the scenario is initialised.  By adding event metaData to the intent we can pull it out at any point during the scenario, rather than only when it's run.

Reply:
Sounds good!